packages/dev/core/src/Materials/Textures/renderTargetTexture.ts

```diff
@@ -902,10 +902,10 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
                     this.renderList.push(mesh);
                 }
             }
-
-            this._waitingRenderList = undefined;
         }
 
+        this._waitingRenderList = undefined;

```

Comment:
```typescript
        if (this._waitingRenderList) {
            if (!this.renderListPredicate) {
                this.renderList = [];
                for (let index = 0; index < this._waitingRenderList.length; index++) {
                    const id = this._waitingRenderList[index];
                    const mesh = scene.getMeshById(id);
                    if (mesh) {
                        this.renderList.push(mesh);
                    }
                }
            }
            this._waitingRenderList = undefined;
        }
```
to avoid setting `this._waitingRenderList = undefined;` each time `render` is called.